src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -2524,6 +2550,9 @@ private uint getJitFlags(ref CORJIT_FLAGS flags, uint sizeInBytes)
 
             flags.corJitFlags2 = CorJitFlag2.CORJIT_FLG2_USE_PINVOKE_HELPERS;
 
+            if (this.MethodBeingCompiled.HasCustomAttribute("System.Runtime.InteropServices", "NativeCallableAttribute"))
```

Comment:
Done. Works very nicely for the synthetic Main method - that needs to be nativecallable as well.